curriculum/challenges/english/25-front-end-development/lab-blog-post-card/66eaddd04a9e533fba689001.md

```diff
@@ -0,0 +1,403 @@
+6. The `post-content` class should add padding inside the card for spacing.
+7. The `.post-title` and `.post-excerpt` elements within `.post-content` should have a text color other than the default and a margin on all sides.
+8. The `.read-more` element should be styled like a button and have:
+
+   - A text color other than the default.
+   - Background color.
+   - Margin on all sides.
+   - A `display` property of `inline-block`.
+   - Rounded corners.
+   - Padding on all sides.
+
+9. The `.read-more` element should have a hover effect that changes the background color.
+
+**Note:** Be sure to link your stylesheet in your HTML and apply your CSS.
+
+# --hints--
+
+You should have a `div` with a class of `blog-post-card`.
+
+```js
+assert.exists(document.querySelector("div.blog-post-card"));
+```
+
+You should have a an `img` element with a `class` of `post-img`. Make sure your image has an `alt` text and `src` value.
+
+```js
+const card = document.querySelector("div.blog-post-card");
+assert.exists(card);
+
+const img = card.querySelector("img.post-img");
+assert.exists(img);
+assert.isString(img.alt);
+assert.isNotEmpty(img.alt);
+assert.isTrue(img.hasAttribute("src"));
+assert.isNotEmpty(img.getAttribute("src"));
+```
+
+You should have a `div` with class of `post-content` .
+
+```js
+const card = document.querySelector("div.blog-post-card");
+assert.exists(card);
+const content = card.querySelector("div.post-content");
+assert.exists(content);
+```
+
+You should have an `h2` with a `class` of `post-title`. Make sure it has some text content.
+
+```js
+const card = document.querySelector("div.blog-post-card");
+assert.exists(card);
+
+const content = card.querySelector("div.post-content");
+assert.exists(content);
+
+const postTitle = content.querySelector("h2.post-title");
+assert.exists(postTitle);
+assert.isString(postTitle.textContent);
+assert.isNotEmpty(postTitle.textContent);
+```
+
+You should have a `p.post-excerpt`.
+
+```js
+const card = document.querySelector("div.blog-post-card");
+assert.exists(card);
+
+const content = card.querySelector("div.post-content");
+assert.exists(content);
+
+const postExcerpt = content.querySelector("p.post-excerpt");
+assert.exists(postExcerpt);
+```
+
+You should have an `a` with a `class` of `read-more`.
+
+```js
+const card = document.querySelector("div.blog-post-card");
+assert.exists(card);
+
+const content = card.querySelector("div.post-content");
+assert.exists(content);
+
+const readMore = content.querySelector("a.read-more");
+assert.exists(readMore);
+```
+
+Your `a.read-more` should have a `Read More` text.
+
+```js
+const card = document.querySelector("div.blog-post-card");
+assert.exists(card);
+
+const content = card.querySelector("div.post-content");
+assert.exists(content);
+
+const link = content.querySelector("a.read-more");
+assert.exists(link);
+assert.strictEqual(link.textContent, "Read More");
```

Comment:
According to instructions, the order of elements isn't strict.
The updated test would allow the campers to place the `a` tag in any order within the `post-content ` `div`.

```suggestion
const el = document.querySelector("div.post-content a.read-more");
assert.exists(el);
assert.strictEqual(el.textContent, "Read More");
```